executor/adapter.go

```diff
@@ -759,13 +762,32 @@ func (a *ExecStmt) logAudit() {
 // FormatSQL is used to format the original SQL, e.g. truncating long SQL, appending prepared arguments.
 func FormatSQL(sql string, pps variable.PreparedParams) stringutil.StringerFunc {
 	return func() string {
+		if len(pps) > 0 {
+			sql = formatPreparedSQL(sql, pps)
+		}
 		cfg := config.GetGlobalConfig()
 		length := len(sql)
 		if maxQueryLen := atomic.LoadUint64(&cfg.Log.QueryLogMaxLen); uint64(length) > maxQueryLen {
 			sql = fmt.Sprintf("%.*q(len:%d)", maxQueryLen, sql, length)
 		}
-		return QueryReplacer.Replace(sql) + pps.String()
+		return QueryReplacer.Replace(sql)
+	}
+}
+
+// formatPreparedSQL is used to replace the prepared params for original SQL
+func formatPreparedSQL(sql string, pps variable.PreparedParams) string {
+	sqlBuffer := bytes.NewBuffer([]byte{})
+	paramsIndex := 0
+	for i := 0; i < len(sql); i++ {
+		c := sql[i]
+		if c != '?' {
+			sqlBuffer.WriteByte(c)
+			continue
+		}
+		sqlBuffer.WriteString(fmt.Sprintf("%v", pps[paramsIndex].GetValue()))
```

Comment:
I think we should handle the format of null value and string value, for example, add quotes around a string(`abc`->`"abc"`).
You may take a reference of  `types.DatumsToString` .

Reply:
i will fix it